### PR TITLE
Doc String for Learner & ConvLearner

### PR DESCRIPTION
**Pull request by Holisticnature**
@jph00 I made a first attempt to document Learner & ConvLearner. Some of these I wish had a little more detail in their naming conventions. While this is [not C++](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules), I do like the rule of thumb of keeping names descriptive where possible. However, this does follow the general style of other ML libraries, so maybe just getting clarification on the docstrings is enough. 
I did wanted to check on tmp_name. I was not sure exactly what this was storing? Is it activations/weights? Temporary files?

### CODE REVIEW

File: fastai/learner.py [hunk 1]
```diff
@@ -14,6 +14,18 @@
...
+        """
+        Learner is a class that combines a ModelData object with a nn.Module object, such that you can train that
+        module.
+        data (ModelData): An instance of ModelData.
+        models: chosen neural architecture for solving a supported problem.
+                Eg. ResNet-34
+        opt_fn: optimizer function, uses SGD with Momentum of .9 if none.
+        tmp_name: output name of the directory containing temporary files from training process
+        models_name: output name of the directory containing the trained model
+        metrics(function): some function for evaluating a desired metric. Eg. accuracy.
```
Comment (jph00) [inline fastai/learner.py]: It's an array of functions.

File: fastai/learner.py [hunk 2]
```diff
@@ -14,6 +14,18 @@
 
 class Learner():
     def __init__(self, data, models, opt_fn=None, tmp_name='tmp', models_name='models', metrics=None, clip=None):
+        """
+        Learner is a class that combines a ModelData object with a nn.Module object, such that you can train that
```
Comment (jph00) [inline fastai/learner.py]: Remove "Learner is a class that"

File: fastai/learner.py [hunk 3]
```diff
@@ -14,6 +14,18 @@
...
+        module.
+        data (ModelData): An instance of ModelData.
+        models: chosen neural architecture for solving a supported problem.
+                Eg. ResNet-34
```
Comment (jph00) [inline fastai/learner.py]: 'ResNet-34' would not be a valid input here. If giving an example of param use, be sure it's a valid input for that param.

File: fastai/learner.py [hunk 4]
```diff
@@ -14,6 +14,18 @@
...
+        opt_fn: optimizer function, uses SGD with Momentum of .9 if none.
+        tmp_name: output name of the directory containing temporary files from training process
+        models_name: output name of the directory containing the trained model
+        metrics(function): some function for evaluating a desired metric. Eg. accuracy.
+        clip: gradient clip chosen to limit the change in the gradient to prevent exploding gradients
```
Comment (jph00) [inline fastai/learner.py]: It would be helpful to say what data type this is or give an example

File: fastai/learner.py (hunk 3 above)
Comment (Holisticnature) [reply fastai/learner.py]: Interesting. I was trying to mimic a similar param I saw in the library. I think this makes sense, I might have misread the example took from. Thanks for clarification.

File: fastai/learner.py (hunk 2 above)
Comment (Holisticnature) [reply fastai/learner.py]: Redundant. Understood.

File: fastai/learner.py (hunk 1 above)
Comment (Holisticnature) [reply fastai/learner.py]: Thanks!

File: fastai/learner.py (hunk 3 above)
Comment (Holisticnature) [reply fastai/learner.py]: As a point of clarification, models represents a nn.Module correct?

**Pull request merged**